stdlib/public/core/StringObject.swift

```diff
@@ -467,6 +489,12 @@ extension _StringObject {
     _internalInvariant(isLarge)
     return (discriminatedObjectRawBits & 0x4000_0000_0000_0000) != 0
   }
+  
+  @inline(__always)
+  internal var isImmortalCocoa: Bool {
```

Comment:
This would implies more than just "immortal" and "cocoa", but also shared (i.e. fast-UTF).